examples/torch/ddpg_example/controllers/supervisor/models/networks.py

```diff
@@ -0,0 +1,291 @@
+
+        self.optimizer = optim.Adam(self.parameters(), lr=lr)
+
+        self.device = T.device('cuda:0' if T.cuda.is_available() else 'cpu')
+
+        self.to(self.device)
+
+    def forward(self, state, action):
+        state_value = self.fc1(state)
+        state_value = self.bn1(state_value)
+        state_value = F.relu(state_value)
+
+        state_value = self.fc2(state_value)
+        state_value = self.bn2(state_value)
+
+        action_value = F.relu(self.action_value(action))
+        state_action_value = F.relu(T.add(state_value, action_value))
+
+        state_action_value = self.q(state_action_value)
+
+        return state_action_value
+
+    def save_checkpoint(self):
+        print("...saving checkpoint....")
+        T.save(self.state_dict(), self.checkpoint_file)
+
+    def load_checkpoint(self):
+        print("..loading checkpoint...")
+        self.load_state_dict(T.load(self.checkpoint_file))
+
+
+class ActorNetwork(nn.Module):
+    def __init__(self,
+                 lr,
+                 input_dims,
+                 fc1_dims,
+                 fc2_dims,
+                 n_actions,
+                 name,
+                 chkpt_dir='tmp/ddpg'):
+        super(ActorNetwork, self).__init__()
+        self.lr = lr
+        self.input_dims = input_dims
+        self.fc1_dims = fc1_dims
+        self.fc2_dims = fc2_dims
+        self.n_actions = n_actions
+        self.checkpoint_file = os.path.join(chkpt_dir, name + "_ddpg")
+
+        self.fc1 = nn.Linear(*self.input_dims, self.fc1_dims)
+        self.bn1 = nn.LayerNorm(self.fc1_dims)
+
+        self.fc2 = nn.Linear(self.fc1_dims, self.fc2_dims)
+        self.bn2 = nn.LayerNorm(self.fc2_dims)
+        self.mu = nn.Linear(self.fc2_dims, self.n_actions)
+
+        self.initialization()
+
+        self.optimizer = optim.Adam(self.parameters(), lr=lr)
+        self.device = T.device('cuda:0' if T.cuda.is_available() else 'cpu')
+        self.to(self.device)
+
+    def initialization(self):
+        f1 = 1 / np.sqrt(self.fc1.weight.data.size()[0])
+        T.nn.init.uniform_(self.fc1.weight.data, -f1, f1)
+        T.nn.init.uniform_(self.fc1.bias.data, -f1, f1)
+
+        f2 = 1 / np.sqrt(self.fc2.weight.data.size()[0])
+        T.nn.init.uniform_(self.fc2.weight.data, -f2, f2)
+        T.nn.init.uniform_(self.fc2.bias.data, -f2, f2)
+
+        f3 = 0.0003
+        T.nn.init.uniform_(self.mu.weight.data, -f3, f3)
+        T.nn.init.uniform_(self.mu.bias.data, -f3, f3)
+
+    def forward(self, state):
+        x = self.fc1(state)
+        x = self.bn1(x)
+        x = self.fc2(x)
+        x = self.bn2(x)
+        x = T.tanh(self.mu(x))
+
+        return x
+
+    def save_checkpoint(self):
+        print("...saving checkpoint....")
+        T.save(self.state_dict(), self.checkpoint_file)
+
+    def load_checkpoint(self):
+        print("..loading checkpoint...")
+        self.load_state_dict(T.load(self.checkpoint_file))
+
+
+class Agent(object):
```

Comment:
Agent class is basically the DDPG algorithm. I think it shouldn't be inside the networks.py file